satellite/satellitedb/certdb.go

```diff
@@ -21,15 +21,20 @@ type certDB struct {
 
 func (certs *certDB) SavePublicKey(ctx context.Context, nodeID storj.NodeID, publicKey crypto.PublicKey) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	_, err = certs.db.Get_CertRecord_By_Id(ctx, dbx.CertRecord_Id(nodeID.Bytes()))
-	if err == sql.ErrNoRows {
-		return certs.tryAddPublicKey(ctx, nodeID, publicKey)
+	pubbytes, err := pkcrypto.PublicKeyToPKIX(publicKey)
+	if err != nil {
+		return Error.Wrap(err)
 	}
+
+	_, err = certs.db.Get_CertRecord_By_Publickey(ctx, dbx.CertRecord_Publickey(pubbytes))
```

Comment:
This does an expensive check.

Reply:
updated with upsert type concept but used transactions. Only thing little different is check if same public key exists, if it then just exit no UPDATE is done

Reply:
will address this comment in https://github.com/storj/storj/pull/2704/ PR

Reply:
Upsert is different from transactions.